File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/dayview/DayViewEvent.java
package com.wiseassblog.samsaradayplanner.ui.dayview;

/**
 * It models interactions between the View and the Logic class
 */
public class DayViewEvent {

    private final Event event;
    private final Object value;

    public DayViewEvent(Event event, Object value){
        this.event = event;
        this.value = value;
    }

    public Event getEvent() {
        return event;
    }

    public Object getValue() {
        return value;
    }

    public enum Event {
        ON_START,
        ON_HOUR_SELECTED,
        ON_MANAGE_TASKS_SELECTED
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/dayview/DayViewEvent.kt
package com.wiseassblog.samsaradayplanner.ui.dayview

/**
 * It models interactions between the View and the Logic class
 */
sealed class DayViewEvent {
    object OnStart : DayViewEvent()
    data class OnHourSelected(val position: Int) : DayViewEvent()
    object OnManageTasksSelected : DayViewEvent()

}